spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/store/jwk/JwkSetConverter.java

```diff
@@ -137,13 +137,15 @@ private JwkDefinition createRsaJwkDefinition(Map<String, String> attributes) {
 		}
 
 		// use
-		JwkDefinition.PublicKeyUse publicKeyUse =
-				JwkDefinition.PublicKeyUse.fromValue(attributes.get(PUBLIC_KEY_USE));
-		if (!JwkDefinition.PublicKeyUse.SIG.equals(publicKeyUse)) {
-			throw new JwkException((publicKeyUse != null ? publicKeyUse.value() : "unknown") +
-					" (" + PUBLIC_KEY_USE + ") is currently not supported.");
+		JwkDefinition.PublicKeyUse publicKeyUse = null;
+		if (attributes.containsKey(PUBLIC_KEY_USE)) {
```

Comment:
I see this check will bypass the validation, but will this default to "sig" or "enc" when validating the signature.

Reply:
According to spec it's optional, they did not specify any default value. Also I went through the code the PublicKeyUse is not used any where in the code.

Reply:
@senkum @Yasinjama Although the spec does say the `use` parameter is optional, it is required by `JwkTokenStore` because it only supports signature (`sig`)  verification and NOT payload decryption (`enc`).

If we make this optional in `JwkTokenStore` that how do we know if the JWK is meant for `sig` or `enc`?

Reply:
@jgrandja, I went through the code base , PublicKeyUse is not used any where in the code. 

Can we default to sig when PublicKeyUse is not present?

Reply:
@senkum PublicKeyUse is used during the validation of the JWK(s) after received from the JWK Set Endpoint. This is required as the JWK can be used for `sig` or `enc`. We cannot default to `sig`. If the returned JWK is meant for `enc` and we default to `sig` this would introduce a bug.

Reply:
@jgrandja I may be wrong, but when I searched the entire code base, 

The only use of the PublicKeyUse i found are in the following places
   - JwkSetConverter in the methods createRsaJwkDefinition and createEllipticCurveJwkDefinition, the validation is done, this is where exception is thrown when the value is null.
   - JwkSetConverterTest tests for JwkSetConverter
   - JwkDefinition - member variable, this variable is not used anywhere

Nowhere else it's used, please correct me if I am wrong

Reply:
@senkum You are correct, `PublicKeyUse` is used/required in:

> JwkSetConverter in the methods createRsaJwkDefinition and createEllipticCurveJwkDefinition, the validation is done, this is where exception is thrown when the value is null.

As mentioned in my previous comment, this validation is required because it's possible that one of the JWK is meant for `enc`. Therefore, we need to distinguish if the JWK is meant for `sig` or `enc`. This is why we require the `use` parameter. We cannot default to `sig` for all returned JWK's as this would introduce a bug if the JWK is actually meant for `enc`.

Reply:
@senkum I'm going to close this PR based on the reason I mentioned in my previous [comment](https://github.com/spring-projects/spring-security-oauth/pull/1348#discussion_r191760386).

Reply:
This should not have been closed: the implementation does not follow the spec, so this is a bug.
Now, for example, Spring security does not work with the following JWK key set that is perfectly valid but that does not specify a use attribute:
`{ "keys": [
    { "kty": "RSA",
      "x5t#S256": "...",
      "e": "...",
      "x5t": "...",
      "kid": "SIGNING_KEY",
      "x5c": [
        "...",
        "..."
      ],
      "key_ops": [
        "encrypt",
        "verify"
      ],
      "alg": "RS256",
      "n": "..."
    }
  ]
}`
And unfortunately, it seems impossible to adapt the `JwkSetConverter`.
This should be re-opened / re-considered.

Reply:
which artifact version has got this fix?